### PR TITLE
148 Update RabbitMQ to standards

### PR DESCRIPTION
**Pull request by JasonBock**
## Description
Updated the project to standards (and added a test project)

### CODE REVIEW

File: RockLib.Messaging.RabbitMQ.Tests/RabbitReceiverTests.cs [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+using RabbitMQ.Client;
+using RabbitMQ.Client.Events;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace RockLib.Messaging.RabbitMQ.Tests
+{
+    public static class RabbitReceiverTests
```
Comment (NeilSchiefer) [inline RockLib.Messaging.RabbitMQ.Tests/RabbitReceiverTests.cs:12]: 👍

File: RockLib.Messaging.RabbitMQ.Tests/RabbitReceiverTests.cs [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+            var connection = new Mock<IConnection>(MockBehavior.Strict);
+            connection.Setup(_ => _.CreateModel()).Returns(model.Object);
+            connection.Setup(_ => _.Dispose());
+
+            var connectionFactory = new Mock<IConnectionFactory>(MockBehavior.Strict);
+            connectionFactory.Setup(_ => _.CreateConnection()).Returns(connection.Object);
+
+            using (var receiver = new RabbitReceiver("name", connectionFactory.Object, "queue", "exchange",
+                new List<string>() { "routingKey" }, 20, true))
+            {
```
Comment (NeilSchiefer) [inline RockLib.Messaging.RabbitMQ.Tests/RabbitReceiverTests.cs:61]: Looks like the RoutingKeys is public probably could assert that its setup correctly.

File: RockLib.Messaging.RabbitMQ.Tests/RabbitSenderTests.cs [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+            var connection = new Mock<IConnection>(MockBehavior.Strict);
+            connection.Setup(_ => _.Dispose());
+
+            var connectionFactory = new Mock<IConnectionFactory>(MockBehavior.Strict);
+            connectionFactory.Setup(_ => _.CreateConnection()).Returns(connection.Object);
+
+            using (var sender = new RabbitSender("name", connectionFactory.Object,
+                "exchange", "routingKey", "routingKeyHeaderName", false))
+            {
+                _ = sender.Connection;
```
Comment (NeilSchiefer) [inline RockLib.Messaging.RabbitMQ.Tests/RabbitSenderTests.cs:66]: What does this give us?  Not understanding why the assignment to a variable we don't care about.

Comment (JasonBock) [reply RockLib.Messaging.RabbitMQ.Tests/RabbitSenderTests.cs:66]: It's a little weird, but there's a `Lazy<IConnection>` underneath the scenes, so to get the disposal logic tested in the right way, I need to read the property, but I don't need to do anything with it. Hence the use of the discard.

File: RockLib.Messaging.RabbitMQ.Tests/RabbitReceiverTests.cs (hunk 2 above)
Comment (JasonBock) [reply RockLib.Messaging.RabbitMQ.Tests/RabbitReceiverTests.cs:63]: Added asserts for that now.

**Review by eslutz — APPROVED** [review]

File: RockLib.Messaging.RabbitMQ.Tests/RabbitSenderTests.cs (hunk 3 above)
Comment (NeilSchiefer) [reply RockLib.Messaging.RabbitMQ.Tests/RabbitSenderTests.cs:66]: 👍   Cool figured I was missing something.

**Review by NeilSchiefer — APPROVED** [review]

**Pull request merged**